ts/client/services/syncer.ts

```diff
@@ -81,13 +139,47 @@ export class SyncerService {
         console.info(`#${batchID}  [${Usage[usage]}]`, batch.toString());
     }
 
+    getPubkeyFromTxn(txn: { data: string }): Pubkey {
+        // Get public key from registration call data
+        const { args } = this.accountRegistry.interface.parseTransaction(txn);
+        return new Pubkey(args.pubkey);
+    }
+
+    async handleSinglePubkeyRegistered(event: Event) {
+        const pubkeyID = event.args?.pubkeyID;
+
+        const txn = await event.getTransaction();
+        const pubkey = this.getPubkeyFromTxn(txn);
+
+        await this.pubkeyStorage.update(pubkeyID, pubkey);
+        await this.pubkeyStorage.commit();
+
+        console.info(`Pubkey added ID ${pubkeyID} ${pubkey.toString()}`);
+    }
+
+    async handleBatchPubkeyRegistered(_event: Event) {
+        console.error("handleBatchPubkeyRegistered not implemented");
+    }
+
     newBatchListener = async (
         batchID: null,
         accountRoot: null,
         batchType: null,
         event: Event
     ) => {
-        return await this.handleNewBatch(event);
+        return this.handleNewBatch(event);
+    };
+
+    singlePubkeyRegisteredListener = async (pubkeyID: null, event: Event) => {
+        return this.handleSinglePubkeyRegistered(event);
+    };
+
+    batchPubkeyRegisteredListener = async (
+        startID: null,
+        endID: null,
+        event: Event
+    ) => {
+        return this.handleSinglePubkeyRegistered(event);
```

Comment:
Should be handleBatchPubkeyRegistered here?
```suggestion
        return this.handleBatchPubkeyRegistered(event);
```

Reply:
:facepalm: Yes, good catch